threescale/src/structs.rs

```diff
@@ -252,3 +252,24 @@ impl Message {
         }
     }
 }
+
+// Info relevant for the RateLimit headers.
+#[derive(Clone)]
+pub struct RateLimitInfo {
+    /// Header RateLimit-Limit's value.
+    pub limit: u64,
+    /// Header RateLimit-Remaining's value.
+    pub remaining: u64,
+    /// Header RateLimit-Reset's value.
+    pub reset: u64,
+}
+
+impl Default for RateLimitInfo {
+    fn default() -> Self {
+        RateLimitInfo {
+            limit: u64::MAX,
+            remaining: u64::MAX,
+            reset: u64::MAX,
```

Comment:
If any one of these is never set then it will have a maximum u64 value, which is not what we want. This is asking for you to use an Option and default it all to None, then emit headers when they are `Some(...)`.

Reply:
I was thinking Optioning the whole struct at first but thought about too many unwraps and checks but option for fields is better! Added.